tests/core/test_holders.py

```diff
@@ -109,6 +110,16 @@ def test_get_memory_usage_with_trace():
     assert_equal(memory_usage['nb_requests_by_array'], 1.25)  # 15 calculations / 12 arrays
 
 
+def test_set_input_dispath_by_period():
+    simulation = get_simulation(single)
+    variable = simulation.tax_benefit_system.get_variable('housing_occupancy_status')
+    entity = simulation.household
+    holder = Holder(variable, entity)
+    set_input_dispatch_by_period(holder, make_period(2019), 'owner')
```

Comment:
```suggestion
    set_input_dispatch_by_period(holder, make_period(2019), np.asarray(['owner']))
```

It seems needlessly confusing to pass a scalar value to a method whose signature makes it clear it expects an array, and the test's pass/fail status does not as far as I can tell depend on that.

It could also make sense to use a non-Enum variable for this test, as they are the exception rather than the norm in OpenFisca, and using an exceptional case hints that the behaviour under test is restricted to that exceptional case.